private/buf/bufwork/bufwork.go

```diff
@@ -94,31 +94,31 @@ var (
 	}
 )
 
-// NewWorkspace returns a new workspace.
-func NewWorkspace(
-	ctx context.Context,
-	config *Config,
-	readBucket storage.ReadBucket,
+// WorkspaceBuilder builds workspaces.
+type WorkspaceBuilder interface {
+	// BuildWorkspace builds a bufmodule.Workspace for the given targetSubDirPath.
+	BuildWorkspace(
+		ctx context.Context,
+		workspaceConfig *Config,
+		readBucket storage.ReadBucket,
+		relativeRootPath string,
+		targetSubDirPath string,
+		configOverride string,
+		externalDirOrFilePaths []string,
+		externalDirOrFilePathsAllowNotExist bool,
```

Comment:
If we feel strongly about these positional parameters, we could roll out an `BuildWorkspaceOption` for each one (e.g. `BuildWorkspaceWithConfigOverride`, etc.).

Reply:
I think that it makes sense to take an options approach, especially given the nature of these arguments. I am a big +1 on that, but I don't think it blocks this change either.

Reply:
Eh yea this is a historical relic, but also would be fine if we kept for now and dealt with this in `bufwire` at the same time

Reply:
Sounds good, keeping as-is for now.